tests/helper/helper_documentation.go

```diff
@@ -0,0 +1,108 @@
+package helper
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+	"unicode"
+
+	. "github.com/onsi/gomega"
+)
+
+const (
+	timePatternInOdo = `(\[[0-9smh]+\])` // e.g. [4s], [1m], [3ms]
+	staticTimeValue  = "[1s]"
+	// Credit: https://github.com/acarl005/stripansi/blob/master/stripansi.go
+	ansiPattern          = "[\u001B\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;[a-zA-Z\\d]*)*)?\u0007)|(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PRZcf-ntqry=><~]))"
+	unicodeSpinnerFrames = "◓◐◑◒"
+)
+
+// ReplaceAllTimeInString replaces the time taken to download a Devfile or a starter project for an odo command with a custom value;
+// this function is helpful because the time value is variable and replacing it with the value in mdx content helps in comparing.
+func ReplaceAllTimeInString(docString string, timeString string) string {
+	reg := regexp.MustCompile(timePatternInOdo)
+	return reg.ReplaceAllString(docString, timeString)
+}
+
+// StripSpinner strips the cmd out string of spaces, spinner statements and spinner frames
+func StripSpinner(docString string) (returnString string) {
+	for _, line := range strings.Split(docString, "\n") {
+		// trim any special character present in the line
+		line = strings.TrimFunc(line, unicode.IsSpace)
+		// This check is to avoid spinner statements in the cmd output
+		// currently it does so for init and dev
+		// e.g. " •  Syncing file changes ..."
+		if (strings.HasPrefix(line, "•  Downloading") || strings.HasPrefix(line, "•  Syncing") || strings.HasPrefix(line, "•  Building")) && strings.HasSuffix(line, "...") {
+			continue
+		}
+
+		// for some reason, splitting the docString by \n does not split the spinner frames,
+		// so we perform a side operation to remove the extra spinner frames that are not present in the final output
+		// e.g. "◐  Downloading devfile "java-maven" from registry "DefaultDevfileReg... \n ◓  Downloading devfile "java-maven" from registry "DefaultDevfileReg...\n\n [32m✓[0m  Downloading devfile "java-maven" from registry "DefaultDevfileRegistry" [2s]"
+		if strings.ContainsAny(line, unicodeSpinnerFrames) {
+			line = "✓" + strings.SplitAfter(line, "✓")[1]
+		}
+
+		returnString += line + "\n"
+	}
+	// replace all instances of time to [1s], this is also done for mdx out
+	returnString = ReplaceAllTimeInString(returnString, staticTimeValue)
+	return
+}
+
+// GetMDXContent reads the content of MDX files, strips it of extra spaces and returns the string
+// it strips the extra space for an easy comparison
+func GetMDXContent(filePath string) (mdxContent string) {
+	// filename of this file
+	_, filename, _, _ := runtime.Caller(0)
+	// path to the docs directory
+	mdxDir := filepath.Join(filepath.Dir(filename), "..", "..", "docs", "website", "docs")
+
+	readFile, err := os.Open(filepath.Join(mdxDir, filePath))
+	defer func(err error) {
+		if err == nil {
+			readFile.Close()
+		}
+	}(err)
```

Comment:
Is there a particular reason you are doing this, instead of a `defer readFile.Close()` after `Expect(err).ToNot(HaveOccurred())` as usual?

Reply:
Yes, gosec throws an error if I simply defer to closing the file.
```sh
make sec 
go run -mod=vendor github.com/securego/gosec/v2/cmd/gosec -severity medium -confidence medium -exclude G304,G204,G107 -quiet  ./tests/integration/... ./tests/helper... ./tests/e2escenarios/... ./tests/documentation/...
Results:


[/home/pvala/go/src/github.com/redhat-developer/odo/tests/helper/helper_documentation.go:71] - G307 (CWE-703): Deferring unsafe method "Close" on type "*os.File" (Confidence: HIGH, Severity: MEDIUM)
    70:         Expect(err).ToNot(HaveOccurred())
  > 71:         defer readFile.Close()
    72: 



Summary:
  Gosec  : dev
  Files  : 23
  Lines  : 3221
  Nosec  : 3
  Issues : 1

exit status 1
make: *** [Makefile:114: sec] Error 1
```